### PR TITLE
Replace `var` with `final` keyword & Fix typo

### PR DESCRIPTION
**Pull request by Juliotati**
A minor clean-up to replace all the `var` keywords with `final` within the three files.
In special cases where final doesn't apply, the datatype of the variable was used instead.

Typo fixed inside `engine.dart`
Before `postion`
After `position`

### CODE REVIEW

File: lib/engine.dart [hunk 1]
```diff
@@ -123,16 +121,12 @@ class Piece {
   Iterable<Delta> get deltas sync* {
     switch (type) {
       case PieceType.king:
-        for (int x = -1; x <= 1; ++x) {
-          for (int y = -1; y <= 1; ++y) {
-            for (int d = 1; d <= kMoveRange; ++d) {
```
Comment (abarth) [inline lib/engine.dart]: Looks like you're missing kMoveRange.  Maybe a got lost in a rebase?

File: lib/engine.dart [hunk 2]
```diff
@@ -361,56 +354,41 @@ class GameHistory {
...
-    var stake =
+    double winnerRating = currentRating(winner);
+    double loserRating = currentRating(loser);
+    double stake =
         pointsToTransfer(score, expectedScore(winnerRating, loserRating));
     adjustRating(winner, stake);
     adjustRating(loser, -stake);
   }
 
-  void recordRatings(List<Player> alivePlayers, List<Player> deadPlayers) {
```
Comment (abarth) [inline lib/engine.dart:372]: Looks like the new scoring algorithm got dropped in the rebase too.

**Comment by abarth** [comment]
Very cool!  Thanks for the pull request.  It looks like a couple recent changes got lost when you rebased (or merged) your change.  If you clean those up, we'll be able to merge your pull request!

**Comment by Juliotati** [comment]
@abarth I've updated `engine.dart` to include the missing pieces.

**Comment by abarth** [comment]
Looks great!  Thanks so much!

**Pull request merged**